### PR TITLE
PB-3 Project structure

### PR DESCRIPTION
**Pull request by SherlyH221**

### CODE REVIEW

File: pap-frontend/src/app/app.module.ts [hunk 1]
```diff
@@ -3,10 +3,14 @@ import { BrowserModule } from '@angular/platform-browser';
...
 import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
+import { CreateProductComponent } from './product/page/create-product/create-product.component';
+import { ListProductComponent } from './product/page/list-product/list-product.component';
 
 @NgModule({
   declarations: [
-    AppComponent
+    AppComponent,
+    CreateProductComponent,
```
Comment (moolsbytheway) [inline pap-frontend/src/app/app.module.ts]: Create ProductsModule and add angular routing to lazy load the module on route "/products"

File: pap-backend/adapter/build.gradle [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+group 'com.pap'
+version '0.0.1-SNAPSHOT'
+sourceCompatibility = '11'
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
```
Comment (moolsbytheway) [inline pap-backend/adapter/build.gradle]: remove unused dependencies

File: pap-backend/domain/build.gradle [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+group 'com.pap'
+version '0.0.1-SNAPSHOT'
+sourceCompatibility = '11'
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
```
Comment (moolsbytheway) [inline pap-backend/domain/build.gradle]: remove unused dependencies

File: pap-backend/application/build.gradle [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+group 'com.pap'
+version '0.0.1-SNAPSHOT'
+sourceCompatibility = '11'
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
```
Comment (moolsbytheway) [inline pap-backend/application/build.gradle]: remove unused dependencies

**Review by moolsbytheway — CHANGES_REQUESTED** [review]

File: pap-frontend/src/app/app.module.ts (hunk 1 above)
Comment (moolsbytheway) [reply pap-frontend/src/app/app.module.ts]: Les composants createProduct et ProductList doivent être déclarés dans le module ProductsModule. 
Et ajoutés dans le routing de ProductsModule.

**Review by moolsbytheway — CHANGES_REQUESTED** [review]

File: pap-frontend/src/app/app.module.ts (hunk 1 above)
Comment (SherlyH221) [reply pap-frontend/src/app/app.module.ts]: J'ajoute le routing des composants dans le routing de ProductsModule ?

Comment (moolsbytheway) [reply pap-frontend/src/app/app.module.ts]: Oui

Comment (moolsbytheway) [reply pap-frontend/src/app/app.module.ts]: Va sur slack si tu as des questions

File: pap-frontend/src/app/app-routing.module.ts [hunk 5]
```diff
@@ -1,8 +1,7 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
 
-const routes: Routes = [];
-
+const routes: Routes = [{ path: '/products', loadChildren: () => import('./products/products.module').then(m => m.ProductsModule) }];
```
Comment (moolsbytheway) [inline pap-frontend/src/app/app-routing.module.ts]: add a route for "/" that redirects to /products  and another route for pages not found

File: pap-frontend/src/app/products/page/list-products/list-products.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (moolsbytheway) [inline pap-frontend/src/app/products/page/list-products/list-products.component.spec.ts]: remove .spec.ts files we will not be using them

File: pap-frontend/src/app/products/products-routing.module.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { NgModule } from '@angular/core';
+import { RouterModule, Routes } from '@angular/router';
+import { CreateProductComponent } from './page/create-product/create-product.component';
+import { ListProductsComponent } from './page/list-products/list-products.component';
+
+const routes: Routes = [
+  {path:'/create' , component: CreateProductComponent},
+  {path :'/list' , component: ListProductsComponent}
```
Comment (moolsbytheway) [inline pap-frontend/src/app/products/products-routing.module.ts]: and another route for pages not found that renders a component saying that the route is not found, the component should be a shared component shared by the main module and the products module

**Review by moolsbytheway — CHANGES_REQUESTED** [review]

**Review by moolsbytheway — APPROVED** [review]

**Pull request merged**